### PR TITLE
[jmxfetch] bump to 0.26.0 add new critical config options

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Adds new config options for JMXFetch 0.26.0 + bumps the jmxfetch version.

### Motivation

JMXFetch support for concurrent instance metric collection.

### Additional Notes

Do not merge until `0.24.0` release (ie. after https://github.com/DataDog/jmxfetch/pull/199 is merged).

### CODE REVIEW

File: releasenotes/notes/jmxfetch-0.23.0-concurrency-support-14de5c08f927bd1c.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    JMXFetch 0.23.0 introduces support for concurrent instance
+    metric collection. Concurrent collection should avoid stalling
+    healthy instances in the event of issues (networking, system) in 
+    any of the other instances configured. Threenew settings are
```
Comment (olivielpeau) [inline releasenotes/notes/jmxfetch-0.23.0-concurrency-support-14de5c08f927bd1c.yaml]: typo: `Three new`

Comment (olivielpeau) [reply releasenotes/notes/jmxfetch-0.23.0-concurrency-support-14de5c08f927bd1c.yaml]: The detailed description of these settings is easy to understand IMO 👌

I think we should leave it out of the release note though, and rather document this in a more "permanent" location (datadog.yaml for example)

**Review by olivielpeau — COMMENTED** [review]
Left a few comments + I think we'll want to add these options to Agent 5 as well, in the `init_config` section, which means that for backward compatibility we have to make Agent 6 support these option in `init_config` as well

File: pkg/config/config.go [hunk 2]
```diff
@@ -122,6 +122,11 @@ func initConfig(config Config) {
 	config.BindEnvAndSetDefault("hostname_fqdn", false)
 	config.BindEnvAndSetDefault("cluster_name", "")
 
+	// JMXFetch critical settings
```
[the remark this answers is not in the document]
Comment (truthbk) [reply pkg/config/config.go]: 🤦‍♂️

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -324,6 +324,21 @@ api_key:
 # Duration of the restart interval in seconds
 # jmx_restart_interval: 5
 #
+# JMXFetch now collects multiples instances concurrently. The following metrics may
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml]: let's remove the `now`, or it'll still be here in X years ;)

File: pkg/config/config_template.yaml [hunk 4]
```diff
@@ -324,6 +324,21 @@ api_key:
...
+# help fine-tune the level of concurrency and timeouts that come into play during the
+# collection of metrics from configured instances:
+#
+# thread_pool_size: determines the maximum level of concurrency.
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml]: nit: IMO it's a bit weird to write `thread_pool_size` and prefix it with `jmx_` below. Just a description like `# Maximum level of concurrency` would be better I think (ideally with a short explanation of what the user-facing tradeoff would be when it's increased/lowered)

Comment (olivielpeau) [reply pkg/config/config_template.yaml]: (applies to the other options as well)

**Review by olivielpeau — DISMISSED** [review]
A few nits on the config file docs, other than that LGTM!

**Comment by truthbk** [comment]
Had been previously approved - merging after polishing docs.

**Pull request merged**